### PR TITLE
Fix "Sub-directory Support" in small steps

### LINKED ISSUE #12205: "Sub-directory Support" does not accept top-level (root) URL
Summary: when configured following the sub-directory support instructions, paths like `/librenms/devices` and `/librenms/validate` work fine, but the top level path `/librenms/` gives a 404 error.

To reproduce:

* I created a fresh Ubuntu 20.04 lxd container
* I installed LibreNMS git head (2793585c) exactly as per the [Install-LibreNMS](https://docs.librenms.org/Installation/Install-LibreNMS/) page, using Apache rather than Nginx.
* I went through the database setup pages, then validate, then added `localhost` device.  LibreNMS is working fine at this point.
* I then modified the configuration as per [Sub-directory Support](https://docs.librenms.org/Extensions/Sub-Directory/)

```
--- /etc/apache2/sites-available/librenms.conf.old	2020-10-11 17:40:42.714705144 +0000
+++ /etc/apache2/sites-available/librenms.conf	2020-10-11 17:47:59.622870362 +0000
@@ -1,8 +1,10 @@
+AllowEncodedSlashes On
+Alias /librenms /opt/librenms/html
+
 <VirtualHost *:80>
   DocumentRoot /opt/librenms/html/
   ServerName  librenms.example.com

-  AllowEncodedSlashes NoDecode
   <Directory "/opt/librenms/html/">
     Require all granted
     AllowOverride All
```

```
--- /opt/librenms/html/.htaccess.old	2020-10-11 17:45:44.213581077 +0000
+++ /opt/librenms/html/.htaccess	2020-10-11 17:45:51.393649471 +0000
@@ -20,7 +20,7 @@
 </IfModule>

 RewriteEngine on
-RewriteBase /
+RewriteBase /librenms
 RewriteCond %{REQUEST_FILENAME} !-f
 RewriteCond %{REQUEST_FILENAME} !-d
 RewriteCond %{REQUEST_URI} !\.(js|ico|txt|gif|jpg|png|css|php)
```

```
--- /opt/librenms/.env.old	2020-10-11 17:45:25.301400911 +0000
+++ /opt/librenms/.env	2020-10-11 17:45:58.561717747 +0000
@@ -5,5 +5,5 @@
 DB_USERNAME=librenms
 DB_PASSWORD=password

-#APP_URL=
+APP_URL=/librenms/
 NODE_ID=5f83401f8b0fa
```

```
--- /opt/librenms/config.php.old	2020-10-11 17:44:58.193142622 +0000
+++ /opt/librenms/config.php	2020-10-11 17:46:05.021779276 +0000
@@ -8,7 +8,7 @@

 ### This should *only* be set if you want to *force* a particular hostname/port
 ### It will prevent the web interface being usable form any other hostname
-#$config['base_url']        = "/";
+$config["base_url"] = '/librenms/';

 ### Enable this to use rrdcached. Be sure rrd_dir is within the rrdcached dir
 ### and that your web server has permission to talk to rrdcached.
```

```
systemctl restart apache2
systemctl restart php7.4-fpm
```

Now go to path `/librenms/validate` and it's fine:

image

But go to path `/librenms/` and it gives a 404 error:

image

Note that the LibreNMS logo in the top left-hand corner of the screen links to `/librenms/`

image

### PR DESCRIPTION
**Pull request by mpikzink**
This fixes small parts from #12205 part 1
fixes #12205

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Util/Url.php [hunk 1]
```diff
@@ -225,13 +225,7 @@ public static function sensorLink($sensor, $text = null, $type = null, $overlib
      */
     public static function deviceUrl($device, $vars = [])
     {
-        $routeParams = [is_numeric($device) ? $device : $device->device_id];
```
Comment (Jellyfrog) [inline LibreNMS/Util/Url.php:228]: Is this change really working? its used for the device page tabs

Comment (mpikzink) [reply LibreNMS/Util/Url.php:228]: It generates the right Link to the device. But you are right, I haven't tested the tab vars...

Comment (mpikzink) [reply LibreNMS/Util/Url.php:228]: Seems to work.

https://exaple.com/librenms/ports/errors=yes

Screenshot_2021-05-26_20-29-27

Comment (murrant) [reply LibreNMS/Util/Url.php:228]: Here is the problem @mpikzink 
That is the legacy format for device urls... I'm trying to migrate them to the new format.  This basically undoes that and routes all links through the compatibility routes to the device pages.

Comment (mpikzink) [reply LibreNMS/Util/Url.php:228]: @murrant Of course, that's not a good idea on my part then. I don't understand why only this page doesn't work with you, the other pages that have already been converted to Blade work without problems. Currently I have no idea how I can fix the problem differently do you have any idea how I can get further.

**Comment by mpikzink** [comment]
We need to fix this in another way.

**Pull request closed without merging**